subgraphs/bridgeworld/src/mappings/advanced-questing.ts

```diff
@@ -0,0 +1,194 @@
+import { Address, BigInt, log, store } from "@graphprotocol/graph-ts";
+
+import {
+  ADVANCED_QUESTING_ADDRESS,
+  LEGION_ADDRESS,
+  TREASURE_ADDRESS,
+} from "@treasure/constants";
+
+import {
+  AdvancedQuestContinued,
+  AdvancedQuestEnded,
+  AdvancedQuestStarted,
+  AdvancedQuesting,
+  TreasureTriadPlayed,
+} from "../../generated/Advanced Questing/AdvancedQuesting";
+import {
+  AdvancedQuest,
+  AdvancedQuestReward,
+  LegionInfo,
+  Random,
+  TokenQuantity,
+  TreasureTriadResult,
+} from "../../generated/schema";
+import { getAddressId } from "../helpers/utils";
+import { getXpPerLevel } from "../helpers/xp";
+
+export function handleAdvancedQuestStarted(event: AdvancedQuestStarted): void {
+  const params = event.params;
+
+  const quest = new AdvancedQuest(
+    getAddressId(event.address, params._startQuestParams.legionId)
+  );
+
+  quest.requestId = params._requestId;
+  quest.token = getAddressId(LEGION_ADDRESS, params._startQuestParams.legionId);
+  quest.user = params._owner.toHexString();
+  quest.status = "Idle";
+  quest.zoneName = params._startQuestParams.zoneName;
+  quest.part = params._startQuestParams.advanceToPart;
+
+  const treasures: string[] = [];
+
+  for (let i = 0; i < params._startQuestParams.treasureIds.length; i++) {
+    const tokenId = params._startQuestParams.treasureIds[i].toHex();
```

Comment:
This needs to be the full id to a treasure.

```suggestion
    const tokenId = getAddressId(TREASURE_ADDRESS, params._startQuestParams.treasureIds[i]);
```